test/external.c

```diff
@@ -831,6 +831,10 @@ test_write_file_set(hid_t fapl)
     if (reset_raw_data_files(false) < 0)
         TEST_ERROR;
 
+    /* Generate the data */
+    for (i = 0; i < cur_size; i++)
+        whole[i] = (int)i;
+
     /* Create the dataset */
     if ((mem_space = H5Screate_simple(1, &cur_size, &max_size)) < 0)
```

Comment:
If looking to create a reproducer at some point, I get the feeling it may have something to do with the fact that the loop invariant variable `cur_size` is passed into `H5Screate_simple()` when creating the dataspace.